File: practical03/integral.c
#include <stdio.h>
#include <math.h>

int main(void){
    int a,N,i; //initialisation
    a=0;
    N=13;

    float sum,b;
    b=M_PI/3;
    sum=0;

    sum=tan(a)+tan(b); //endpoints

    for(i=1;i<N;i++){
    sum=sum + 2*tan(a+ i*(b-a)/(N-1)); //the argument of tan is each equidistant point, i is the parameter
     printf("%f\n",sum);
    }

    sum=(b-a)*sum/(2*N); //prefactors

    printf("Calucated integral = %f \n", sum);
    printf("Exact solution = %f \n", log(2));
    printf("Difference = %f \n", fabs(sum-log(2)) ); //fabs is the math.h absolute value
    
    
return(0);
}

File: practical03/integral.f90
program integral

    !integer (kind=4) :: N,i
    integer (4) :: N,i
    real(4):: summ, b,a


    real(8),  parameter :: PI  = 4 * atan (1.0_8)

    a=0
    b=PI/3

    summ=0
    N=12

    summ=tan(a)+tan(b)

    do i=2,N-1,1
        summ = summ + 2*tan(a+ i*(b-a)/N)
    end do

    summ=(b-a)*summ/(2*N)

    diff= abs(log(2.) - summ)

    write(6,'(a,f10.8,/,a,f10.8,/,a,f10.8)') 'calculated integral = ',summ,'exact solution = ', log(2.), 'difference = ',diff

end program integral
